### DISCUSSION TITLE
Alert spikes for `ROUTER_EXTERNAL_TARGET_ERROR`

### DISCUSSION THREAD

**Discussion by GMaiolo** [vercel/vercel#7721] [Help]
Hello! We have the Datadog integration with Vercel for our NextJS application, but lately we've been having random spikes of `ROUTER_EXTERNAL_TARGET_ERROR` error reported.

We've arrived to the [Limits docs](https://vercel.com/docs/concepts/limits/overview#proxied-request-timeout) section where they mention this error, which I'll quote below:

> ### Proxied Request Timeout
> 
> The amount of time (in seconds) that a proxied request (`rewrites` or `routes` with an external destination) is allowed to process an HTTP request. The maximum timeout is 30 seconds. If the external server does not reply until the maximum timeout is reached, an error with the message `ROUTER_EXTERNAL_TARGET_ERROR` will be returned.

But as far as I've seen, the request is a NextJS prefetch of the page data made by a [`Link` component](https://nextjs.org/docs/api-reference/next/link)
Here's an example of one of these logs in Datadog:
<img width="718" alt="image" src="https://user-images.githubusercontent.com/9776436/165357670-33720b05-13ba-46dc-91e1-d5713105dac5.png">

The only other noticeable thing is that most of these requests seem to be in spikes and mostly triggered by bots.
For example, according to the `network` object of these Datadog alerts, there are spikes made by `AhrefsBot` and other bots, Crawlers, etc.

Could it be that this `ROUTER_EXTERNAL_TARGET_ERROR` is actually a throttle applied by Vercel to potential crawlers or bots? Or at least is it something fixable from our end?

Thanks!

**Comment by amyegan** [comment]
Hi @GMaiolo. Sorry for the late response. The most common reason we see that error is because of timeouts or unresponsive external resources, like you saw in the docs. I have seen this happen in the past when services like Cloudflare or database platforms are experiencing degraded performance or outages.

I hope this issue has already been resolved for you. But if you still need assistance, please share any additional information you can about your project to help me investigate.

**Reply by GMaiolo** [reply]
Hey @amyegan, thanks for your reply.
Do you think it's safe for us to ignore the error then? It seems to be on Vercel's side, and I see nothing we can do from the application side, is that correct?
